UnitTests/SPStringAdditionsTests.m

```diff
@@ -635,6 +635,18 @@ - (void)testSeparatedIntoLines{
 
 }
 
+- (void)testSeparatedIntoLinesByCharsetObjC{
+
+    NSString *str = @"SELECT * FROM `HKWarningsLog`\n LIMIT 1000;\nSELECT * FROM `HKWarningsLog`\n LIMIT 1000;\nSELECT * FROM `HKWarningsLog` LIMIT 1000\n;";
+    NSArray *expectedArray = @[@"SELECT * FROM `HKWarningsLog`\n LIMIT 1000",@"\nSELECT * FROM `HKWarningsLog`\n LIMIT 1000", @"\nSELECT * FROM `HKWarningsLog` LIMIT 1000\n"];
```

Comment:
Maybe we should trim leading/trailing white space and new lines from the specific query?

Reply:
Gonna go ahead and merge anyways cause it fixes the bug indeed, just a thought here